plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaCloudSlave.java

```diff
@@ -0,0 +1,116 @@
+package io.jenkins.plugins.remotingkafka;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.TaskListener;
+import hudson.slaves.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang.StringUtils;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+public class KafkaCloudSlave extends AbstractCloudSlave {
+    private static final Logger LOGGER = Logger.getLogger(KafkaCloudSlave.class.getName());
+    private static final String DEFAULT_AGENT_PREFIX = "remoting-kafka-agent";
+
+    private String cloudName;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getCloudName() {
+        return cloudName;
+    }
+
+    public void setCloudName(String cloudName) {
+        this.cloudName = cloudName;
+    }
+
+    @Nonnull
+    public KafkaKubernetesCloud getCloud() {
+        return getCloud(cloudName);
+    }
+
+    private static KafkaKubernetesCloud getCloud(String cloudName) {
+        Cloud cloud = Jenkins.get().getCloud(cloudName);
+        if (cloud instanceof KafkaKubernetesCloud) {
+            return (KafkaKubernetesCloud) cloud;
+        } else {
+            throw new IllegalStateException(KafkaCloudSlave.class.getName() + " can be launched only by instances of " + KafkaKubernetesCloud.class.getName() + ". Cloud is " + cloud.getClass().getName());
+        }
+    }
+
+    public String getNamespace() {
+        return getCloud().getNamespace();
+    }
+
+    public KafkaCloudSlave(KafkaKubernetesCloud cloud) throws Descriptor.FormException, IOException {
+        super(getSlaveName(cloud.name),
+                "",
+                // TODO: Remote FS field in Cloud config
+                "/workingDir",
+                1,
+                // TODO: Node usage mode field
+                Mode.NORMAL,
+                cloud.getLabel(),
+                // TODO: SSL field
+                new KafkaComputerLauncher(),
+                // TODO: Retention strat
+                CloudSlaveRetentionStrategy.INSTANCE,
+                // TODO: Node properties field
+                new ArrayList<>());
+
+        this.cloudName = cloud.name;
+    }
+
+    private static String getSlaveName(String baseName) {
+        String randString = RandomStringUtils.random(5, "bcdfghjklmnpqrstvwxz0123456789");
+        String name = baseName;
+        if (StringUtils.isEmpty(name)) {
+            return String.format("%s-%s", DEFAULT_AGENT_PREFIX, randString);
+        }
+        // Because the name is also used in Kubernetes, it should conform to domain name standard
```

Comment:
Is there a way to check if we need another random string because of an agent with the same string? 5 random chars is generally not enough uniqueness